stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -1329,22 +1460,26 @@ public struct Data : ReferenceConvertible, Equatable, Hashable, RandomAccessColl
         if !isKnownUniquelyReferenced(&_backing) {
             _backing = _backing.mutableCopy(_sliceRange)
         }
-        _backing.append(buffer.baseAddress!, length: buffer.count * MemoryLayout<SourceType>.stride)
+        _backing.replaceBytes(in: NSRange(location: _sliceRange.upperBound, length: _backing.length - (_sliceRange.upperBound - _backing._offset)), with: buffer.baseAddress, length: buffer.count * MemoryLayout<SourceType>.stride)
```

Comment:
This seems odd to me from a principled/conceptual standpoint. It seems like the Data should not reaching outside of its `_sliceRange`, and instead that this should be `_backing.replaceBytes(in: NSRange(location: _sliceRange.upperBound, length: 0)…`, which is exactly how you'd implement an abstract "append" if "replace" was the only primitive available.

In practice, I get it. Either:
1. The slice is the last reference of a previously larger backing, and we don't need to keep any bytes after our `_sliceRange` around anymore.
or
2. `_backing` is newly allocated due to CoW and the length parameter ends up 0 anyway.

I feel vaguely like a slightly different abstraction would satisfy both the principled and practiced side, but I'm not sure what that'd be right now.

Do we ever have occasion to clean up the bytes that *precede* the slice range when the slice is the last reference?